arch/arm/core/Kconfig

```diff
@@ -92,6 +92,27 @@ config ARM_NONSECURE_FIRMWARE
 	  resources of the Cortex-M MCU, and, therefore, it shall avoid
 	  accessing them.
 
+menu "ARM Non-Secure Firmware Options"
+depends on ARM_NONSECURE_FIRMWARE
+
+config ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS
+	bool "Non-Secure Firmware uses Secure Entry functions"
+	help
+	  Option indicates that ARM Non-Secure Firmware uses
+	  Secure Entry functions provided by the Secure Firmware.
+	  The Secure Firmware must be configured to provide these functions.
+
+config ARM_ENTRY_VENEERS_LIB_NAME
```

Comment:
Could this be removed, and, instead, re-scope the use of the "same" Kconfig symbol for Secure and Non-Secure firmware?

Reply:
Yes we could, with a little restructuring in the menus (because of Kconfig limitations)

Right now, the existing "ARM Secure Firmware Options" menu wasn't changed. Instead, a new menu was added, so a duplicate menu entry was also needed. If we instead merge the menus, we can avoid the duplication, but the name of the menu must be the same for both secure and non-secure, e.g.

"ARM Secure/Non-Secure Interface Options"
or
"ARM Non-Secure Callable Options"
or
"ARM Secure Entry Function Options"

Reply:
I think we can merge the menus. Can we call the menu: ARM TrustZone Options? Then it should depend on _SECURE_ or _NONSECURE_. And then, the _HAS_ENTRIES shall depend on _SECURE_, the USES_ENTRIES shall depend on _NONSECURE.

It's just a suggestion, though. Maybe @ulfalizer can help-out too?

Reply:
If the issue is making it configurable in two different spots in the menu tree, then a single symbol with two prompts might work:

```
menu "One menu"

config FOO
	bool "Foo support"
	depends on DEP1

endmenu

...

menu "Another menu"

config FOO
	prompt "Foo support"
	depends on DEP2

endmenu
```

(See the note at the end of https://docs.zephyrproject.org/latest/guides/kconfig/index.html#common-shorthands for why it's probably a good idea to use `prompt` instead of `bool` in the second definition.)

Internally, that just creates a single `FOO` symbol, with direct dependencies `DEP1 || DEP2`.

Reply:
That's interesting. We also want just one menu to be visible, though.

Reply:
Right, this is the generic ARM Kconfig source, so we better as little things as possible :)

Reply:
@oyvindronningstad
Should work fine too. The visibility is calculated per menu node (basically, symbols themselves don't have prompts/configurability, only menu nodes).

Reply:
ok, great, this is basically what is being done now. Can we improve it in some way?

Reply:
Can play around with stuff this in the `menuconfig` interface. Should give you an idea how it works:

```
config A
	bool "P1"

config A
	bool "P2"
	depends on DEP

config A
	bool "P3"

config DEP
	bool "dep"
```

Notice that when you toggle any of P1/P2/P3, all them change (because they're actually all the same symbol).

Reply:
We basically want the prompt and default to stay the same, but the depends and help text to change as well as place it in another menu.

Reply:
> ok, great, this is basically what is being done now. Can we improve it in some way?

No problems with using it like that at least. It's an intended feature.

One thing to watch out for is dependencies. Note that they get `OR`ed, so even if one of the definitions has e.g. a `depends on n` (never satisfied), the symbol can still end up as `y` if the dependencies are satisfied at the other location (and it was set to `y` earlier).

Reply:
Ok, so can we remove the default statement from one, and they will still have the same default?

Reply:
Also, can we avoid duplicating the prompt text?

Reply:
Might need to put `default`s in both locations. The reason is that this,

```
config A
	bool
	prompt "A"
	default y
	depends on FOO
```

is really just a shorthand for this:

```
config A
	bool
	prompt "A" if FOO
	default y if FOO
```

Dependencies really attach to properties like `prompt`s and `default`s, and not to entire symbols (ish... a few things care about direct dependencies, but it's obscure). Think of `depends on` as a shorthand for adding `if`s to all the properties at that definition location.

Reply:
> Also, can we avoid duplicating the prompt text?

Nope, will have to duplicate it.

Reply:
Ok, thanks for the explanation @ulfalizer!

Reply:
Do you have an opinion on two (one shown at a time) vs one (merged) menus in this specific case @ulfalizer?

Reply:
If you go into the symbol information in `menuconfig` (or the [online reference](https://docs.zephyrproject.org/latest/reference/kconfig/index.html)), you can see symbol definition(s) with propagated dependencies btw.

Reply:
> Do you have an opinion on two (one shown at a time) vs one (merged) menus in this specific case @ulfalizer?

Will have to check. I only looked at the diff. ;)

Reply:
I agree with @ioannisg that it seems redundant to have both `SECURE` and `NONSECURE`.

I'd also try to move the security options into a shared menu and have just a single definition for `RM_ENTRY_VENEERS_LIB_NAME`.

The two definitions are really close to each other here. I was imagining one of them being off in a menu a bit further away.

Reply:
If you want the user to notice whether it's secure/nonsecure, you could do something like this btw:

```
comment "Secure firmare"
	depends on ARM_SECURE_FIRMWARE

comment "Non-secure firmare"
	depends on !ARM_SECURE_FIRMWARE
```

Those show up in the configuration interface when the dependencies are satisfied.

Reply:
If we cannot aoid having two menus, @oyvindronningstad, @ulfalizer , then, should we, at least, name the ARM_ENTRY_VENEERS_LIB_NAME differently, to avoid confusion?
E.g. in the _SECURE menu, we can call it: _GENERATED_ENTRY_VENEERS_LIB_NAME, and in the _NONSECURE, we can call it: _LINKED/USED_ENTRY_VENEERS_LIB_NAME...

It is still not seen, in this PR, where the symbols is used...

Reply:
Don't know enough about this particular case. Depends on how different they are in what they do I guess. If they're similar enough, then just a single shared symbol might be okay, IMO.

Reply:
I merged the menus now, using @ulfalizer's suggestion with `comment`. Thanks!

Reply:
I like it as it is now.

Reply:
Ok, great